src/EPR.Calculator.API/Services/TransposePomAndOrgDataService.cs

```diff
@@ -130,5 +155,47 @@ public void Transpose(CalcResultsRequestDto resultsRequestDto)
                 }
             }
         }
+
+        private List<OrganisationDetails> GetOrganisationDetailsBySubmissionPeriod(int orgMasterId)
+        {
+            return [.. (from org in context.CalculatorRunOrganisationDataDetails
+                    join pom in context.CalculatorRunPomDataDetails
+                    on new { org.OrganisationId, org.SubmissionPeriodDesc } equals new { pom.OrganisationId, pom.SubmissionPeriodDesc }
+                    where (org.CalculatorRunOrganisationDataMasterId == orgMasterId && org.OrganisationName != null && org.SubsidaryId ==null)
+                    select new OrganisationDetails
+                    {
+                        OrganisationId = org.OrganisationId,
+                        OrganisationName = org.OrganisationName,
+                        SubmissionPeriod = Convert.ToInt32(pom.SubmissionPeriod.Replace("-P", "")),
+                    }).Distinct().OrderByDescending(t => t.SubmissionPeriod)];
+        }
+
+        private List<SubsidaryDetails> GetSubsidaryDetailsBySubmissionPeriod(int orgMasterId)
+        {
+            return (from org in context.CalculatorRunOrganisationDataDetails
+                        join pom in context.CalculatorRunPomDataDetails
+                        on new { org.OrganisationId, org.SubsidaryId, org.SubmissionPeriodDesc } equals new { pom.OrganisationId, pom.SubsidaryId, pom.SubmissionPeriodDesc }
+                        where (org.CalculatorRunOrganisationDataMasterId == orgMasterId && org.OrganisationName != null && org.SubsidaryId != null)
```

Comment:
Should you add a condition to only get RunId specific values?

Reply:
updated as per comment